torch/csrc/jit/passes/common_subexpression_elimination.cpp

```diff
@@ -11,87 +11,110 @@ namespace torch {
 namespace jit {
 namespace {
 
-// The function implements common subexpression elimination.
-// Since the nodes are visited in topological order, one pass is enough.
-void EliminateCommonSubexpression(
-    Block* block,
-    const AliasDb& aliasDb,
-    std::function<Node*(Node*)> parent_lookup_fn) {
-  std::unordered_set<Node*, HashNode, EqualNode> subexprs;
-  for (auto it = block->nodes().begin(); it != block->nodes().end(); ++it) {
-    auto node = *it;
-    if (node->hasSideEffects()) {
-      GRAPH_DEBUG("Node was skipped due to side effects:\n", *node);
-      continue;
-    }
-    if (node->isNondeterministic()) {
-      GRAPH_DEBUG("Node was skipped due to its non determinism:\n", *node);
-      continue;
-    }
-    if (aliasDb.hasWriters(node)) {
-      GRAPH_DEBUG("Node was skipped due to alias analysis result:\n", *node);
-      // Do NOT have enough information to do CSE on these nodes.
-      continue;
-    }
+struct CommonSubexpressionEliminator {
+  CommonSubexpressionEliminator(std::shared_ptr<Graph> graph)
+      : graph_(std::move(graph)) {}
 
-    if (!node->blocks().empty()) {
-      // Traverse sub-blocks.
-      for (auto block : node->blocks()) {
-        EliminateCommonSubexpression(block, aliasDb, [&](Node* n) {
-          auto existing = subexprs.find(n);
-          if (existing != subexprs.end()) {
-            return *existing;
-          }
-
-          return parent_lookup_fn(n);
-        });
+  bool run(std::function<Node*(Node*)> parent_lookup_fn) {
+    return run(graph_->block(), std::move(parent_lookup_fn));
+  }
+
+  // The function implements common subexpression elimination.
+  // Since the nodes are visited in topological order, one pass is enough.
+  bool run(Block* block, std::function<Node*(Node*)> parent_lookup_fn) {
```

Comment:
[Readability]: add a comment to indicate that return true means that the graph has been changed.